include/ocpn_plugin.h

```diff
@@ -172,6 +172,28 @@ class wxGLCanvas;
 //----------------------------------------------------------------------------------------------------------
 //    Some PlugIn API interface object class definitions
 //----------------------------------------------------------------------------------------------------------
+
+/**
+ * Waypoint passing constraint flags for plugin API.
+ * Defines how a vessel should approach and pass a waypoint.
+ * These can be combined with bitwise OR to apply multiple constraints.
+ */
+enum PI_WaypointPassingFlag {
+  /** No specific constraints. */
+  PI_WAYPOINT_PASSING_NONE = 0x00,
```

Comment:
Wouldn't this be better using `enum class PI_WaypointPassingFlag : int {...}`?

Reply:
done